src/api/java/com/minecolonies/api/research/IResearchRequirement.java

```diff
@@ -22,4 +22,11 @@
      * @return translation text component.
      */
     TranslationTextComponent getDesc();
+
+    /**
+     * Get a string description of the requirement, to simplify serialization.
+     * @return string description of the research requirement.
+     */
+    String getAttributes();
```

Comment:
They return a string containing the necessary information to rebuild the user-visible traits for the given research requirement, whatever those may be.

I can get them out of the API methods, if that'd be preferred, since in practice there's not a huge benefit from parsing them versus the traits individually, and most likely only need to be called from within coremod (specifically GlobalResearchFactory).